src/java.desktop/share/classes/javax/imageio/spi/ServiceRegistry.java

```diff
@@ -228,7 +227,7 @@ public Iterator<Class<?>> getCategories() {
     private Iterator<SubRegistry> getSubRegistries(Object provider) {
         List<SubRegistry> l = new ArrayList<>();
         for (Class<?> c : categoryMap.keySet()) {
-            if (c.isAssignableFrom(provider.getClass())) {
+            if (c.isInstance(provider)) {
```

Comment:
Can this be reached if `provider` is null? If yes there could be a change of behaviour as the previous code would have thrown NPE.

Reply:
No. This method is called from 3 places,  and there 3 null checks before the method call.

Reply:
Thanks for double checking! LGTM then.